torchci/pages/hud/[repoOwner]/[repoName]/[branch]/[[...page]].tsx

```diff
@@ -389,14 +388,23 @@ export const PinnedTooltipContext = createContext<[null | string, any]>([
 
 export default function Hud() {
   const router = useRouter();
+  const params = packHudParams(router.query);
 
   // Logic to handle tooltip pinning. The behavior we want is:
   // - If the user clicks on a tooltip, it should be pinned.
   // - While a tooltip is pinned, we don't show any other tooltips.
   // - Clicking outside the tooltip or pressing esc should unpin it.
   // This state needs to be set up at this level because we want to capture all
   // clicks.
-  const [userSettings, setUserSettings] = useState({ useGrouping: true });
+  const [userSettings, setUserSettings] = useState({
+    useGrouping: true,
+  });
+
+  useEffect(() => {
```

Comment:
As mentioned in the top level comment, let's make this so that `useGrouping` is false when any filter is set, not just when the actual URL query string has a filter.